chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/ics/IcsErrorsReportDao.java

```diff
@@ -0,0 +1,82 @@
+package gov.healthit.chpl.scheduler.job.ics;
+
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.dao.impl.BaseDAOImpl;
+import gov.healthit.chpl.entity.CertificationBodyEntity;
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+
+@Repository("icsErrorsReportDao")
+public class IcsErrorsReportDao extends BaseDAOImpl {
+
+    public List<IcsErrorsReportItem> findAll() {
+        List<IcsErrorsReportEntity> entities = this.findAllEntities();
+        return entities.stream()
+                .map(entity -> entity.toDomain())
+                .toList();
+    }
+
+    @Transactional
+    public void deleteAll() {
+        this.findAllEntities().stream()
+        .filter(entity -> !entity.getDeleted())
+        .forEach(entity -> {
+            entity.setDeleted(true);
+            entityManager.merge(entity);
+        });
+        entityManager.flush();
+    }
+
+    @Transactional
+    public void create(List<IcsErrorsReportItem> reports) throws EntityCreationException, EntityRetrievalException {
+        for (IcsErrorsReportItem report : reports) {
+            IcsErrorsReportEntity entity = new IcsErrorsReportEntity();
+            entity.setCertifiedProductId(report.getListingId());
```

Comment:
I feel like these kinds of things have started changing to use builders. Is that not appropriate here?

Reply:
Yes, totally appropriate. I moved the existing file and did not change that part of the code. Updated it now.